### PR TITLE
Allow Group owners to invite users to group

### PR DESCRIPTION
**Pull request by tomekr**
This PR adds the ability for Group owners to invite an existing member to the group via email. Invite acceptance and revocation will come in later PRs. Invites are not generated if the user does not exist in the system or if the recipient already has a pending/unexpired invite.

https://github.com/tomekr/jamroulette/commit/1bbcf39c067ba1fc7aca5868e40a7d5173e2a353 adds the migration and model for an Invite (although this commit was later changed to do a `ago/from_now` bug)
https://github.com/tomekr/jamroulette/commit/83cb7579c79b093fb5d5477f7671527c08553f0e adds an `InvitesController` with a create action that is used to create an invite for a given group, sender, recipient combo and fixes the `ago/from_now` bug mentioned above.

group invite

### CODE REVIEW

File: app/controllers/group_members_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class GroupMembersController < ApplicationController
+  before_action :set_group
+
+  def index
+    authorize! :read, @group
```
Comment (valeksenko) [inline app/controllers/group_members_controller.rb:7]: you should rarely need `authorize!`, prefer `load_and_authorize_resource`

File: app/controllers/invites_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    authorize! :manage, @group
+
+    recipient = User.find_by(email: invite_params[:email])
+    invite = @group.invites.build(
+      sender: current_user,
+      recipient: recipient
+    )
+
+    if invite.save
+      flash[:success] = 'Invite sent!'
```
Comment (valeksenko) [inline app/controllers/invites_controller.rb:16]: `flash.notice` ?

File: app/controllers/invites_controller.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class InvitesController < ApplicationController
+  before_action :set_group
+
+  def create
+    authorize! :manage, @group
+
+    recipient = User.find_by(email: invite_params[:email])
+    invite = @group.invites.build(
```
Comment (valeksenko) [inline app/controllers/invites_controller.rb:10]: we don't use role in the invite?

File: spec/factories/invites.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :invite do
+    group
+    association :recipient, factory: :user
+    association :sender, factory: :user
+
+    invite_token { 'secret-invite-token' }
+    expires_at { Invite::EXPIRES.from_now }
```
Comment (valeksenko) [inline spec/factories/invites.rb:10]: it's a default so could be ommited

File: app/models/group.rb [hunk 5]
```diff
@@ -3,6 +3,7 @@
 class Group < ApplicationRecord
   has_many :group_memberships, dependent: :destroy
   has_many :users, through: :group_memberships
+  has_many :invites, dependent: :destroy
```
Comment (valeksenko) [inline app/models/group.rb:6]: is Rails really smart enough to remove invites where `recipient` and `sender` are set to this user? Specs Or It Didn't Happpen!

File: app/controllers/invites_controller.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  private
+
+  def set_group
+    @group = Group.find(params[:group_id])
+  end
+
+  def invite_params
+    params.require(:invite).permit(:email, :role)
```
Comment (valeksenko) [inline app/controllers/invites_controller.rb:31]: we don't really use these params for mass assignment 🤷

File: config/routes.rb [hunk 7]
```diff
@@ -24,5 +24,8 @@
     get 'random', on: :collection
   end
 
-  resources :groups
+  resources :groups do
+    resources :group_members
+    resources :invites, only: %i[create destroy]
```
Comment (valeksenko) [inline config/routes.rb:29]: `destroy` is going to be implemented later? I'd keep it away until than (and maybe put TODO instead)

File: app/controllers/invites_controller.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      recipient: recipient
+    )
+
+    if invite.save
+      flash[:success] = 'Invite sent!'
+    else
+      flash.alert = invite.errors.full_messages.join(', ')
+    end
+
+    redirect_to group_group_members_path(@group)
```
Comment (valeksenko) [inline app/controllers/invites_controller.rb:21]: that looks pretty ugly, should we use `memberships` name for that controller so it'd be `group_group_membership_path`?

File: app/controllers/invites_controller.rb (hunk 3 above)
Comment (tomekr) [reply app/controllers/invites_controller.rb:10]: The role will be used when invite accepting is implemented

**Pull request merged**